pkg/ivd/ivd_protected_entity_type_manager.go

```diff
@@ -270,32 +274,19 @@ func (this *IVDProtectedEntityTypeManager) copyInt(ctx context.Context, sourcePE
 		if err != nil {
 			return nil, err
 		}
-		keepAfterDeleteVm := true
-
-		vslmCreateSpec := types.VslmCreateSpec{
-			Name:              "ivd-created",
-			KeepAfterDeleteVm: &keepAfterDeleteVm,
-			BackingSpec: &types.VslmCreateSpecDiskFileBackingSpec{
-				VslmCreateSpecBackingSpec: types.VslmCreateSpecBackingSpec{
-					Datastore: md.Datastore,
-				},
-				ProvisioningType: string(types.BaseConfigInfoDiskFileBackingInfoProvisioningTypeThin),
-			},
-			CapacityInMB: md.VirtualStorageObject.Config.CapacityInMB,
-			Profile:      nil,
-			Metadata:     md.ExtendedMetadata,
-		}
-		createTask, err = this.vsom.CreateDisk(ctx, vslmCreateSpec)
+
+		this.logger.Debugf("Ready to provision a new volume with the source Metadata: %v", md)
+		volumeId, err := CreateCnsVolumeInCluster(ctx, this.client, md, this.logger)
```

Comment:
Let's go have CreateCnsVolumeInCluster return an ID rather than a string we then convert with NewIDFromString